holoviews/plotting/mpl/chart.py

```diff
@@ -952,21 +974,30 @@ def _create_bars(self, axis, element, ranges, style):
         # Iterate over group, category and stack dimension values
         # computing xticks and drawing bars and applying styles
         xticks, labels, bar_data = [], [], {}
+        categories = values.get('category', [None])
+        num_categories = len(categories)
         for gidx, grp in enumerate(values.get('group', [None])):
             sel_key = {}
             label = None
             if grp is not None:
                 grp_label = gdim.pprint_value(grp)
                 sel_key[gdim.name] = [grp]
                 yalign = -0.04 if cdim and self.multi_level else 0
-                xticks.append((gidx+0.5, grp_label, yalign))
-            for cidx, cat in enumerate(values.get('category', [None])):
-                xpos = gidx+self.bar_padding+(cidx*width)
+                goffset = width * (num_categories / 2 - 0.5)
+                if num_categories > 1:
+                    # mini offset needed or else combines with non-continuous
+                    goffset += 0.000001
```

Comment:
These kind of magic numbers seems hacky. Are you sure there is not a better way?

Reply:
AFAIK no.

Without:
```python
import numpy as np
import pandas as pd
import holoviews as hv
hv.extension("matplotlib")

samples = 100

pets = ["A", "B"]
genders = [1, 2, 3, 5]

np.random.seed(100)
samples = 100

pets = ['Cat', 'Dog', 'Hamster', 'Rabbit']
genders = ['Female', 'Male', "NA"]

pets_sample = np.random.choice(pets, samples)
gender_sample = np.random.choice(genders, samples)

bars = hv.Bars((pets_sample, gender_sample, np.ones(samples)), ['Pets', 'Gender']).aggregate(function=np.sum)

bars.opts(fig_size=300, aspect=2)
```

<img width="563" alt="image" src="https://github.com/holoviz/holoviews/assets/15331990/8169e9cf-8fb8-4fda-a2df-dca968ef1c14">


With the "hack"
<img width="589" alt="image" src="https://github.com/holoviz/holoviews/assets/15331990/a209bd3e-3a18-429d-887e-efadbae71925">

Reply:
This is what is happening behind the scenes (haven't added a small eps).

``` python
import matplotlib.pyplot as plt
import matplotlib as mpl

# HoloViews
bar_data = {
    "Female": {
        "x": [0.0, 1.0, 2.0, 3.0],
        "height": [6.0, 10.0, 10.0, 7.0],
        "width": 0.26666666666666666,
        "bottom": [0, 0, 0, 0],
        "label": "Female",
        "zorder": 0,
        "edgecolor": "k",
        "color": "#30a2da",
    },
    "NA": {
        "x": [0.26666666666666666, 1.2666666666666666, 2.2666666666666666, 3.2666666666666666],
        "height": [10.0, 10.0, 10.0, 9.0],
        "width": 0.26666666666666666,
        "bottom": [0, 0, 0, 0],
        "label": "NA",
        "zorder": 0,
        "edgecolor": "k",
        "color": "#fc4f30",
    },
    "Male": {
        "x": [0.5333333333333333, 1.5333333333333332, 2.533333333333333, 3.533333333333333],
        "height": [8.0, 7.0, 6.0, 7.0],
        "width": 0.26666666666666666,
        "bottom": [0, 0, 0, 0],
        "label": "Male",
        "zorder": 0,
        "edgecolor": "k",
        "color": "#e5ae38",
    },
}
xaxis_kwargs = [
    (0.26666666666666666, "Cat", -0.04),
    (0.0, "Female", 0),
    (0.26666666666666666, "NA", 0),
    (0.5333333333333333, "Male", 0),
    (1.2666666666666666, "Rabbit", -0.04),
    (1.0, "Female", 0),
    (1.2666666666666666, "NA", 0),
    (1.5333333333333332, "Male", 0),
    (2.2666666666666666, "Hamster", -0.04),
    (2.0, "Female", 0),
    (2.2666666666666666, "NA", 0),
    (2.533333333333333, "Male", 0),
    (3.2666666666666666, "Dog", -0.04),
    (3.0, "Female", 0),
    (3.2666666666666666, "NA", 0),
    (3.533333333333333, "Male", 0),
]


fig, axis = plt.subplots()
bars = [getattr(axis, "bar")(**bar_spec) for bar_spec in bar_data.values()]
ticks, labels, alignments = zip(*sorted(xaxis_kwargs, key=lambda x: x[0]))
axis.set_xticks(ticks)
axis.set_xticklabels(labels)
for t, y in zip(axis.get_xticklabels(), alignments):
    t.set_y(y)

```

I would have thought I could use `t.set_text(label)` in the for-loop. But it does not seem to work with Matplotlib, see here: 
``` python
import matplotlib.pyplot as plt

fig, axis = plt.subplots()
for t in axis.get_xticklabels():
    t.set_text('a')
```

So the small diff is OK. Maybe make it relative to the width: `goffset += width / 1000`